test/suites/integration/instance_tag_test.go

```diff
@@ -0,0 +1,181 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package integration_test
+
+import (
+	"fmt"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/samber/lo"
+	v1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/aws/karpenter-core/pkg/apis/v1alpha5"
+	corev1beta1 "github.com/aws/karpenter-core/pkg/apis/v1beta1"
+	coretest "github.com/aws/karpenter-core/pkg/test"
+	"github.com/aws/karpenter/pkg/apis/settings"
+	"github.com/aws/karpenter/pkg/apis/v1alpha1"
+	"github.com/aws/karpenter/pkg/apis/v1beta1"
+	"github.com/aws/karpenter/pkg/providers/instance"
+	"github.com/aws/karpenter/pkg/test"
+)
+
+// TODO: Update NodeClaim tag expectation with name when nodeclaim expectation becomes available
+var _ = Describe("EC2 Instance Tagging", func() {
+	var nodeClass *v1beta1.EC2NodeClass
+	var nodePool *corev1beta1.NodePool
+
+	BeforeEach(func() {
+		Skip("NodeClaim tagging tests disabled until v1beta1")
+
+		nodeClass = test.EC2NodeClass(v1beta1.EC2NodeClass{Spec: v1beta1.EC2NodeClassSpec{
+			SecurityGroupSelectorTerms: []v1beta1.SecurityGroupSelectorTerm{{
+				Tags: map[string]string{"karpenter.sh/discovery": settings.FromContext(env.Context).ClusterName},
+			}},
+			SubnetSelectorTerms: []v1beta1.SubnetSelectorTerm{{
+				Tags: map[string]string{"karpenter.sh/discovery": settings.FromContext(env.Context).ClusterName},
+			}},
+		}})
+
+		nodePool = coretest.NodePool(corev1beta1.NodePool{
+			Spec: corev1beta1.NodePoolSpec{
+				Template: corev1beta1.NodeClaimTemplate{
+					Spec: corev1beta1.NodeClaimSpec{
+						NodeClass: &corev1beta1.NodeClassReference{
+							Name: nodeClass.Name,
+						},
+						Requirements: []v1.NodeSelectorRequirement{
```

Comment:
This seems like a lot of requirements to be setting for this test case. Is there a reason that you needed to set this many requirements to get the test working?

Reply:
Nope, probably not. Honestly don't remember why they were there at this point but default requirements should be fine.